Correction of the Spanish translations

programandala-net opened the pull request:
I've reviewed, corrected and completed the Spanish translations, which still had issues, language errors, and missing and misplaced texts. I know other changes are more subjective and depend on preferences.

During the process both versions, <es.po> and <es_ES.po>, became more and more similar by combining their best contents. In fact I worked by diff'ing them on Vim, and with the POT template as well. Eventually it made little sense to keep both files just to preserve a few synonyms, some small style/syntax variants or even a couple of English words and shortcuts, while one single translation would be enough, understandable by any Spanish speaker in any country and easier to mantain.

That's the reason <es_ES.po> is removed. Of course its credits have been preserved in <es.po>.

Since <es.po> is complete and correct now, it seems a good starting point in case others find it useful to create regional variants. Another option is to improve <es.po> to make it as much generic Spanish as possible, without regional specific expressions and terms.

I hope previous translators find this change reasonable and useful.

Beside, some format checking would be appreciated. I had not edited a PO file in years. It seems fine according to the specifications, but I didn't try to compile the program yet.

Minoru commented on po/es.po:
```diff
@@ -1514,24 +1505,23 @@ msgstr "Descargar"
-#, fuzzy
 msgid "Mark as Finished"
-msgstr "Purgar finalizadas"
+msgstr "Marcar como terminada/s"
```
The slash looks suspicious to me, is this correct?

(I don't speak Spanish, feel free to ignore)

Minoru commented on po/es.po:
```diff
@@ -1559,29 +1552,29 @@ msgstr "%sCargando %s..."
 #, c-format
 msgid "Error while retrieving %s: %s"
-msgstr "Error al obtener %s: %s"
+msgstr "Error al descargar %s: %s."
```
Ditto about punctuation.

Minoru commented on po/es.po:
```diff
@@ -1559,29 +1552,29 @@ msgstr "%sCargando %s..."
 #: src/reloader.cpp:125
 msgid "Error: invalid feed!"
-msgstr "Error: ¡fuente inválida!"
+msgstr "Error: fuente incorrecta."
```
Ditto about punctuation.

Minoru replied:
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -1559,29 +1552,29 @@ msgstr "%sCargando %s..."
 #, c-format
 msgid "Error: unsupported URL: %s"
-msgstr "Error: URL no soportada: %s"
+msgstr "Error: URL no soportada: %s."
```
Ditto about punctuation.

Minoru commented on po/es.po:
```diff
@@ -1597,40 +1590,40 @@ msgstr "atributo no encontrado"
 #: src/urlviewformaction.cpp:48 src/urlviewformaction.cpp:61
 msgid "No link selected!"
-msgstr "¡Ningún enlace seleccionado!"
+msgstr "Ningún enlace seleccionado."
```
Ditto about punctuation.

Minoru replied:
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -1597,40 +1590,40 @@ msgstr "atributo no encontrado"
 #: src/urlviewformaction.cpp:162
 msgid "URLs"
-msgstr "URLs"
+msgstr "URL"
```
"Les URL", perhaps? (See above about similar string in ConfigContainer)

Minoru replied:
Withdrawn since titles don't need articles, see https://github.com/newsboat/newsboat/pull/874#issuecomment-612441172.

Minoru commented on po/es.po:
```diff
@@ -1597,40 +1590,40 @@ msgstr "atributo no encontrado"
 #: src/view.cpp:493 src/view.cpp:519
 msgid "Error: feed contains no items!"
-msgstr "Error: ¡la fuente no contiene elementos!"
+msgstr "Error: la fuente no contiene elementos."
```
Ditto about punctuation.

Minoru replied:
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -1662,48 +1655,22 @@ msgstr "no se encontró ningún canal RSS"
-"para UID %u!"
+msgstr "Error fatal: no se pudo determinar el directorio del usuario.\n"
+"Por favor, define la variable de entorno HOME o añade un usuario válido "
+"para UID %u."
```
Ditto about punctuation (exclamation mark replaced by period).

Minoru replied:
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

programandala-net commented:
Thank you for the checking.

> Translations get relatively little attention, so don't expect this to
> be reviewed by any other Spanish speaker any time soon. I'll still
> merge this, of course.

Glad to help. The translation has been much improved. All language
errors, orthographic oversights, etc., are fixed, and missing
translations have been added.

Of course, beside a few typos, some more subtle details need to be
confirmed, like the ones you noticed about the punctuation, which
depends on the usage of the original strings, not always clear from the
source. 

> You replaced a lot of exclamation marks with periods—why? Is
> exclamation considered a very extreme thing in Spanish, not used for
> mere error messages?

Exactly. In English, in this context, exclamations marks are very
common. In Spanish they aren't. In mere error messages, which in fact
are just informative texts, exclamation marks look and sound strange, as
if the text cries out or blames the reader... People use them in
software errors just influenced by English, but they don't look so
natural in Spanish.

> Please make changes as additional commits, without force-pushes,
> otherwise it'd be hard for me to see what you changed. I can do a
> squash before merging, or you can do a rebase if you want—we can
> negotiate that.

> -# OmeGa <omega@mailoo.org>, 2013.
> +# ROOT <epsilon@correoe.no-ip.org>, 2008-2010 (es_ES.po)

> Please put all these names into `Language-Team` field below. Keep the
> emails, drop copyrights and everything else. That's the preferred
> style throughout all translations now, we just didn't bother with
> updating all the files; doing it as opportunity shows up instead.

Thanks. I didn't know the convention, so I listed them as usual in
source code.

>  #: newsboat.cpp:122
> -#, fuzzy
> +#. An actual filename, no translation needed
> 
> If it was a filename, we wouldn't have added it to the template ;)
> Instead, this is a description of the file used to print out a table
> like this:

Yes, it was strange to me, an actual filename. But I checked the source
and didn't see other context.

> ```
> Files:
> 	- config:  /home/minoru/.newsboat/config
> 	- urls:    /home/minoru/.newsboat/urls
> 	- cache:   /home/minoru/.newsboat/cache.db
> ```

Now it's clear, thanks. Fixed. I also added translators notes.

When I checked the code I didn't remember the actual file is "cache.db",
but anyway the original "urls" was misleading as the description of the
"urls" file, instead of "URLs".

>  #: src/configcontainer.cpp:305
>  msgid "%N %V - URLs"
> -msgstr "%N %V - URLs"
> +msgstr "%N %V - URL"

> Is this correct? This string is a dialog title; the dialog contains a
> list of the URLs from inside the article. Perhaps this can be
> translated some other way, e.g. "Las URL" (judging from other
> translations, "las" is the plural article in Spanish)?

In Spanish, acronyms like "URL" are invariable, they can not be
pluralized in written form (reference rule, in Spanish:
https://www.rae.es/dpd/sigla#3). Therefore, when they are used in plural
in a sentence, the plural article is used (in this case the feminine one
"las").

My doubt here was the actual context and usage, which didn't seem clear
to me from the source code. In case "URL" is displayed out of a
sentence, as a tag, in a status line, a heading, etc., the article
should not be used. May you confirm? Thank you.

>  #: src/configparser.cpp:112
>  #, c-format
>  msgid "Error while processing command `%s' (%s line %u): %s"
> -msgstr "Error al procesar el comando `%s' (%s línea %u): %s"
> +msgstr "Error al procesar el comando `%s' (%s línea %u): %s."

> I'm not ready to explain why the original message doesn't end with a
> period, but please don't add, remove, or change punctuation unless
> absolutely necessary.

There are several similar cases. I had searched the PO file for all
"error:" `msgid` with and without period at the end, and after checking
some of them in the source code it seemed there was no reason why
periods were missing in some cases. Beside, many previous translated
strings added or omitted the period in Spanish, regardless of the
original. Therefore it seemed there were oversights not only in the
original but also in both Spanish translations.

My conclusion was all of the original errors that start with "Error:"
were full sentences, so their translations should be punctuated as such.

But as you say, the original punctuaction should be preserved, just in
case, even if it seems wrong, until it's confirmed or corrected.

Nevertheless, what about the translations where the Spanish exclamation
marks have been removed? In those cases, there's almost no doubt the
English original is a full sentence, which no more text added after it
at runtime. Therefore, as you noted, I used a period.

>  #: src/controller.cpp:420
>  msgid "Prepopulating query feeds..."
> -msgstr "Rellenando las fuentes de consulta..."
> +msgstr "Prerrellenando las fuentes de consulta... "

> I understand the original looks a bit weird, but please keep the
> translations in line with the original. Submit a PR against the code
> to fix it there, and it can then trickle down into translations ;)

Sorry, I don't understand what you mean. Could you elaborate?

By the way, I just see the trailing space, which is wrong. I added and
removed many of them that didn't match the original texts, but I oversaw
this one.

> There's now two spaces between "Error:" and the following text, is this intentional?

Of course not. Thanks. Two such typos of mine.

>  #: src/itemlistformaction.cpp:832
>  msgid "No item selected!"
> -msgstr "¡Ningún elemento seleccionado!"
> +msgstr "No hay elementos seleccionados."
> 
>  #: src/dialogsformaction.cpp:98
>  msgid "Error: you can't remove the feed list!"
> -msgstr "Error: ¡no puedes eliminar la lista de fuentes!"
> +msgstr "Error: no puedes eliminar la lista de fuentes."
> 
> Ditto about punctuation.

As said about replacing exclamation marks, these cases, as used in the
source code, looked complete sentences. Therefore a period was needed at
the end. It's even more clear in "Error:" messages.

The question is: If the original text would not have exclamation marks,
would it have a period instead? It depends on the actual usage of the
string and its context.

> >  msgid "%s: %s: unknown command"
> -msgstr "comando inválido `%s'"
> +msgid "%s: %s: comando desconocido"
> 
> This should say `msgstr`, not `msgid`. This line is the reason all the builds fail, see [this log](https://cirrus-ci.com/task/4977138625413120?command=build#L129) for example.

Of course. Sorry. Fixed.

> +msgstr "ptasn"
> 
> This seems incorrect: the original string has six characters, the translation has five.
> 
>  "¿Ordenar por (p)rimera etiqueta/(t)ítulo/número de (a)rtículos/número de "
> 
> Looks like "(l)astupdated" wasn't added to this translation.

Both errors are related: That option and its key were missing in the old
translation files and I didn't realize that. It seems a new addition,
since, I check some translations now, they are missing also from fr.po
and br_PT.po, though not from it.po and de.po.

By the way, in these kind of shortcut legends, the English original uses
kind of tags or variable names. Some translations use longer, clear,
full text descriptions (fr.pt, es.pt), while others use a format similar
to the original (it.po, de.po, pt_PT.po)... Which is the prefered one?

> +#. An actual tag, no translation needed.
> 
> This isn't a tag, it's a description of the type of the content.
> 
> >  msgid "video"
>  msgstr ""

Thanks. Same case about the supposed "actual files" before. Fixed.

>  msgid ""
>  "Please set the HOME environment variable or add a valid user for UID %u!"
>  msgstr ""
> -"¡Por favor define la variable de entorno HOME, o añade un usuario válido "
> -"para UID %u!"
> +"Por favor, define la variable de entorno HOME o añade un usuario válido "
> +"para UID %u."
> 
> Ditto about punctuation (I'm ok with a comma after "Por favor", but the exclamation mark should probably be kept as-is)

One of the previous translations used comma and the other one didn't. In
written Spanish, a comma is better in those cases, though not mandatory.
Punctuation depends on the language, the rules are not identical in all
languages. As for the Spanish exclamation marks, as said, they have no
reason in these kind of messages.

> >  msgid "Mark as Finished"
> -msgstr "Purgar finalizadas"
> +msgstr "Marcar como terminada/s"
> 
> The slash looks suspicious to me, is this correct?

As you see, the old translation was wrong: it read "purge" instead of
"mark", and in plural (i.e. purge several finished items). The other
Spanish translation is preserved, even with the slash.

The issue here is the English participle doesn't change in number, and
it was not clear from the source code whether the command can be applied
to several selected entries or just one. I use the program, but don't
know all its options and possible configurations. That's why I preserved
the optional plural notation "/s" of the previous translator.

Is that option used always with a single entry? Then the "/s" part must
be removed.

>  #: src/urlviewformaction.cpp:162
>  msgid "URLs"
> -msgstr "URLs"
> +msgstr "URL"
> 
> "Les URL", perhaps? (See above about similar string in ConfigContainer)

Same case, as you say. May you explain the actual context/usage of this
string? 

Thank you for your help.

-- 
Marcos Cruz
http://programandala.net

Minoru commented:
Re exclamation marks replaced by periods: got it. I withdraw those comments, then; let's keep these periods.

Re other punctuation changes: I'm not convinced that single-sentence error messages should end with a period. I never gave this much thought. I'll have to asses the current situation and find some style guides before making a decision (I'll file a reminder issue about this). Either way, translations should follow the original; so please remove the periods you've added.

> The question is: If the original text would not have exclamation marks, would it have a period instead? It depends on the actual usage of the string and its context.

I hope to answer that in that reminder issue I'll file.

> but anyway the original "urls" was misleading as the description of the "urls" file, instead of "URLs".

Fair point. I think this should be fixed. I'll submit a PR. Thanks!

> >  #: src/configcontainer.cpp:305
> >  msgid "%N %V - URLs"
> > -msgstr "%N %V - URLs"
> > +msgstr "%N %V - URL"
> 
> My doubt here was the actual context and usage, which didn't seem clear to me from the source code. In case "URL" is displayed out of a sentence, as a tag, in a status line, a heading, etc., the article should not be used. May you confirm? Thank you.

It's indeed a dialog title. Okay, so no article then. Thanks for the explanation!

> >  #: src/controller.cpp:420
> >  msgid "Prepopulating query feeds..."
> > -msgstr "Rellenando las fuentes de consulta..."
> > +msgstr "Prerrellenando las fuentes de consulta... "
> >
> > I understand the original looks a bit weird, but please keep the translations in line with the original. Submit a PR against the code to fix it there, and it can then trickle down into translations ;)
>
> Sorry, I don't understand what you mean. Could you elaborate?
>
> By the way, I just see the trailing space, which is wrong. I added and removed many of them that didn't match the original texts, but I oversaw this one.

I meant the trailing space. I thought you added it intentionally, to add a gap in between "Prepopulating query feeds..." and "done". Sorry for a confusing comment there!

> > +msgstr "ptasn"
> > 
> > This seems incorrect: the original string has six characters, the translation has five.
> > 
> >  "¿Ordenar por (p)rimera etiqueta/(t)ítulo/número de (a)rtículos/número de "
> > 
> > Looks like "(l)astupdated" wasn't added to this translation.
>
> […] It seems a new addition,

Yeah, I think it was added within the last year.

> […] since, I check some translations now, they are missing also from fr.po and br_PT.po, though not from it.po and de.po.

Yeah, but note that translations in fr.po and pt_BR.po are marked as "fuzzy". They simply weren't updated yet.

> By the way, in these kind of shortcut legends, the English original uses kind of tags or variable names. Some translations use longer, clear, full text descriptions (fr.pt, es.pt), while others use a format similar to the original (it.po, de.po, pt_PT.po)... Which is the prefered one?

I assume it's just how things are translated to other languages. I don't speak any of them, but I know that in German they like to string words together to look like a single word.

I'd prefer the legend to be as short as possible, since it has to fit onto the screen (we don't have word-wrapping).

> >  msgid "Mark as Finished"
> > -msgstr "Purgar finalizadas"
> > +msgstr "Marcar como terminada/s"
> > 
> > The slash looks suspicious to me, is this correct?
>
> The issue here is the English participle doesn't change in number […]
>
> Is that option used always with a single entry? Then the "/s" part must be removed.

Yes, it's always used with a single download in Podboat.

> >  #: src/urlviewformaction.cpp:162
> >  msgid "URLs"
> > -msgstr "URLs"
> > +msgstr "URL"
> > 
> > "Les URL", perhaps? (See above about similar string in ConfigContainer)
> 
> Same case, as you say. May you explain the actual context/usage of this string? 

It's basically the same as "%N %V - URLs" above, just used in a different context. Newsboat [lets users have multiple dialogs open at once](https://newsboat.org/releases/2.19/docs/newsboat.html#_advanced_dialog_management), and switch between them. When user presses <kbd>v</kbd>, they see a list of currently open dialogs. That's where this string is shown. Unlike "%N %V - URLs", it doesn't contain the program version and is not user-configurable.

I think it should have the same translation as "%N %V - URLs", i.e. just "URL", without an article or anything.

Minoru commented on po/es.po:
```diff
@@ -431,14 +431,14 @@ msgid "%u unread articles"
 #, c-format
 msgid "Error: couldn't open configuration file `%s'!"
-msgstr "Error: ¡no se pudo abrir el archivo de configuración `%s'!"
+msgstr "Error:  no se pudo abrir el archivo de configuración `%s'"
```
Since exclamation marks are converted into periods, this message should end with a period too.

Minoru commented on po/es.po:
```diff
@@ -459,20 +459,20 @@ msgstr "Cerrar diálogo"
 #: src/itemlistformaction.cpp:832
 msgid "No item selected!"
-msgstr "¡Ningún elemento seleccionado!"
+msgstr "No hay elementos seleccionados."
```
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -459,20 +459,20 @@ msgstr "Cerrar diálogo"
 #: src/dialogsformaction.cpp:98
 msgid "Error: you can't remove the feed list!"
-msgstr "Error: ¡no puedes eliminar la lista de fuentes!"
+msgstr "Error: no puedes eliminar la lista de fuentes."
```
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -459,20 +459,20 @@ msgstr "Cerrar diálogo"
 #: src/itemlistformaction.cpp:1261 src/urlviewformaction.cpp:151
 msgid "Invalid position!"
-msgstr "¡Posición inválida!"
+msgstr "Posición incorrecta."
```
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -550,46 +550,43 @@ msgstr "Elementos compartidos"
 #: src/feedlistformaction.cpp:334
 msgid "No feed selected!"
-msgstr "¡Ninguna fuente seleccionada!"
+msgstr "No hay fuentes seleccionadas."
```
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -550,46 +550,43 @@ msgstr "Elementos compartidos"
 #: src/feedlistformaction.cpp:255
 msgid "Cannot open query feeds in the browser!"
-msgstr ""
+msgstr "No puedo abrir las fuentes de consulta en el navegador."
```
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -683,7 +680,7 @@ msgstr "Ayuda"
 #: src/feedlistformaction.cpp:922 src/itemlistformaction.cpp:818
 msgid "Error: couldn't parse filter command!"
-msgstr "Error: ¡no se pudo procesar el comando de filtro!"
+msgstr "Error: no se pudo procesar el comando de filtrado."
```
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -692,15 +689,15 @@ msgstr "Buscando..."
 #: src/feedlistformaction.cpp:972 src/itemlistformaction.cpp:1256
 msgid "Position not visible!"
-msgstr "¡La posición no es visible!"
+msgstr "Posición no visible."
```
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -827,69 +822,70 @@ msgid "embedded flash"
 #: src/itemlistformaction.cpp:393 src/itemlistformaction.cpp:1289
 msgid "Error: no item selected!"
-msgstr "Error: ¡ningún elemento seleccionado!"
+msgstr "Error:  ningún elemento seleccionado."
```
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -1083,7 +1078,7 @@ msgstr "Artículo - %s"
 #: src/itemviewformaction.cpp:648
 msgid "Error: invalid regular expression!"
-msgstr "Error: ¡expresión regular inválida!"
+msgstr "Error:  expresión regular incorrecta."
```
Comment regarding punctuation is withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -1415,86 +1409,83 @@ msgstr "el atributo `%s' no está disponible."
 #: src/pbcontroller.cpp:145
 msgid "Fatal error: couldn't determine home directory!"
-msgstr "Error fatal: ¡no se pudo determinar el directorio de inicio!"
+msgstr "Error fatal: no se pudo encontrar el directorio de usuario."
```
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

Minoru commented on po/es.po:
```diff
@@ -1415,86 +1409,83 @@ msgstr "el atributo `%s' no está disponible."
-"¡Por favor define la variable de entorno HOME, o añade un usuario válido "
-"para UID %u!"
+"Por favor, define la variable de entorno HOME o añade un usuario válido "
+"para UID %u."
```
Withdrawn, [see explanation in a comment below](https://github.com/newsboat/newsboat/pull/874#issuecomment-612408710).

programandala-net commented:
> I'm not convinced that single-sentence error messages should end with
> a period. I never gave this much thought. I'll have to asses the
> current situation and find some style guides before making a decision
> (I'll file a reminder issue about this). Either way, translations
> should follow the original; so please remove the periods you've added.

Agreed.

In order to check the current situation and decide what to do, consider
these remarks:

If error messages are used as full sentences, they should end with a
period. This is even clearer considering that in theory an error message
could contain more than one sentence, say:

```
Error: Something is broken. Please restart your system.
```

Currently all of the error messages that don't end with a period are
those that end with a paren or a parameter. Therefore a possible
explanation is the parameter always includes the period. But an
oversight seems more likely.

In fact there are 3 messages that end with a parameter and a period.
See, for example:

```
#: src/controller.cpp:627
#, c-format
msgid "An error occurred while parsing %s."
msgstr "Ocurrió un error al procesar %s."
```

When the ending punctuation is included in the parameter, a note for
translators would be helpful, to make sure the punctuation is always
coherent.

See this one, with a paren:

```
#: src/controller.cpp:162 src/controller.cpp:220 src/pbcontroller.cpp:255
#, c-format
msgid "Error: an instance of %s is already running (PID: %u)"
msgstr "Error: una instancia de %s ya está ejecutándose (PID: %u)"
```

Maybe there's a reason in the code for the missing period, I don't know.
Anyway, if I recall clearly, in English the period should be before the
closing paren. In Spanish it must be always after it.

> I'd prefer the legend to be as short as possible, since it has to fit
> onto the screen (we don't have word-wrapping).

I'll think how to reword them to save space.

> > Is that option used always with a single entry? Then the "/s" part
> > must be removed.
> 
> Yes, it's always used with a single download in Podboat.

Thanks. Fixed.

> When user presses <kbd>v</kbd>, they see a
> list of currently open dialogs. That's where this string is shown.
> Unlike "%N %V - URLs", it doesn't contain the program version and is
> not user-configurable.

I see. Then it's correct, no article.

By the way, I see the legend of the dialogs list displays "ENTER" as a
key name, which is not included in the translation. It should be added
when possible. In fact its Spanish name "Intro" is already used in one
message.

-- 
Marcos Cruz
http://programandala.net

Minoru commented:
I copied punctuation remarks to the issue I filed about this, #879. We can continue that particular discussion there.

> By the way, I see the legend of the dialogs list displays "ENTER" as a key name, which is not included in the translation. It should be added when possible. In fact its Spanish name "Intro" is already used in one message.

In that context, it's an identifier, not a name. Other similar identifiers are HOME, END, PPAGE etc. — all listed in the docs. Users write those into the config, so they shouldn't be translated.

programandala-net commented:
> I copied punctuation remarks to the issue I filed about this, #879. We can continue that particular discussion there.

Thanks. I was about to do the same when I saw it!

> > the legend of the dialogs list displays "ENTER" as a key name, which
> > is not included in the translation.
> 
> In that context, it's an identifier, not a name. Other similar
> identifiers are HOME, END, PPAGE etc. — all listed in the docs. Users
> write those into the config, so they shouldn't be translated.

OK, thanks.

-- 
Marcos Cruz
http://programandala.net

Minoru commented:
@programandala-net, you said you've addressed some of my comments, but I don't see any new commits. Are you working on some more changes, or did you simply forgot to push?

programandala-net commented:
> @programandala-net, you said you've addressed some of my comments, but
> I don't see any new commits. Are you working on some more changes, or
> did you simply forgot to push?

I did all the fixes and changes we discussed. I wanted to review
everything one more time, just in case, but had to postpone a bit
because of a project of mine.

Also, there are some punctuation issues that never were properly
adapted: usage of quotes and dashes.

For example this:

  msgid "Search Result - '%s'"

It should be something like:

  msgstr "Resultado de la búsqueda: «%s»"

But sometimes it depends on the actual relation between the text and the
variable, what must be checked in the source code.

I'm not sure whether doing these changes now or in a second phase.

-- 
Marcos Cruz
http://programandala.net

Minoru commented:
> I wanted to review everything one more time, just in case, but had to postpone a bit because of a project of mine.

Ah, ok. Just making sure this PR didn't stall.

> I'm not sure whether doing these changes now or in a second phase.

I'd finish this one, merged it, then submitted a new PR once you got those quotes sorted out. That approach guarantees progress, and even if you get distracted, Newsboat is still left with a somewhat-improved translation. And when you get back to it, smaller PR is easier to rebase onto current master :)

Seems like you're trying to merge Newsboat's master instead of rebasing onto it. Please don't do this, it makes the history convoluted and hard to follow. Have you committed your changes yet? Do you need help with rebasing?

programandala-net commented:
> Seems like you're trying to merge Newsboat's master instead of
> rebasing onto it. Please don't do this, it makes the history
> convoluted and hard to follow.

Sorry, I didn't know that. Yes, I updated my local repo with `pull
upstream master` in order to compile the latest sources and also update
my origin. 

>Have you committed your changes yet?

No, I didn't commit the last version of the translation to my local repo
yet.

>Do you need help with rebasing?

Yes, it's been some time since I collaborated this way. Please tell me
the usual steps. Thank you.

-- 
Marcos Cruz
http://programandala.net

Minoru commented:
> Yes, I updated my local repo with `pull upstream master` in order to compile the latest sources and also update my origin.

Got it. To avoid this problem in the future, you can create branches for each PR you do. That'd allow you to keep your `master` up-to-date without modifying the PR branches.

If you need a fresh upstream master in your local clone, simply check out `upstream/master`. You won't be able to push it to GitHub (because that'd conflict with this PR), but at least you can build latest versions.

Back to the rebase. I'm not sure if you actually need it, so let's start with a simpler solution: roll your `master` back a bit, commit your changes, and push that:

1. make sure we're on `master`:

        $ git checkout master

2. make sure the working copy is clean. The following command should *not* show any modified files:

        $ git status

3. roll `master` back to the last non-merge commit:

        $ git reset --hard ff1f69a8ffdaad5c881fba41eda62564f49518aa

4. restore your uncommitted changes from backup and commit them, then push as usual:

        $ git commit -a
        $ git push

Minoru commented:
Ah, sorry, the last command in my previous comment should've said `git push --force`. I've updated the comment.

programandala-net commented:
> To avoid this problem in the future, you can create branches for each
> PR you do. That'd allow you to keep your `master` up-to-date without
> modifying the PR branches.
> 
> If you need a fresh upstream master in your local clone, simply check
> out `upstream/master`. You won't be able to push it to GitHub (because
> that'd conflict with this PR), but at least you can build latest
> versions.

Thank you, I understand.

>         $ git checkout master
>         $ git reset --hard ff1f69a8ffdaad5c881fba41eda62564f49518aa
> 
> 4. restore your uncommitted changes from backup and commit them, then
> push as usual:
> 
>         $ git commit -a
>         $ git push --force

Done. Anything else?

-- 
Marcos Cruz
http://programandala.net

Minoru commented on po/es.po:
```diff
@@ -121,19 +116,19 @@ msgid "Files:"
 
 #: newsboat.cpp:122
-#. An actual filename, no translation needed
+#. TRANSLATORS: This is the 'config' filename description used in a table.
```
This is going to be removed the next time someone runs `make msgmerge`. Please add these comments to the source code. They should start with three forward slashes for gettext to pick them up.

For these comments to be added to .po files, you'll have to run `make extract && make msgmerge`, but please don't do that: it will create conflicts with current master branch. I'll do this after merging your PR, and then you can start your next PR with an updated version.

Just like before, I've commented about this in a few more places to make it easier to track.

Minoru commented on po/es.po:
```diff
@@ -121,19 +116,19 @@ msgid "Files:"
 
 #: newsboat.cpp:123
-#. An actual filename, no translation needed
+#. TRANSLATORS: This is the 'urls' filename description used in a table.
```
Ditto about comments.

Minoru commented on po/es.po:
```diff
@@ -121,19 +116,19 @@ msgid "Files:"
 
 #: newsboat.cpp:124
-#. An actual filename, no translation needed
+#. TRANSLATORS: This is the 'cache' filename description used in a table.
```
Ditto about comments.

Minoru commented on po/es.po:
```diff
@@ -229,12 +233,12 @@ msgstr "%N %V - Seleccionar etiqueta"
 #: src/configcontainer.cpp:305
 msgid "%N %V - URLs"
-msgstr "%N %V - URLs"
+msgstr "%N %V - URL"
```
Withdrawing this because titles don't need articles, see https://github.com/newsboat/newsboat/pull/874#issuecomment-612441172.

Minoru commented on po/es.po:
```diff
@@ -822,12 +817,10 @@ msgid "embedded flash"
 #: src/htmlrenderer.cpp:987
-#. An actual tag, no translation needed.
 msgid "audio"
 msgstr ""
```
Did you forget to add a translation here? You translated "video" above, after all.

Minoru commented on po/es.po:
```diff
@@ -1520,8 +1513,9 @@ msgid "Play"
 msgstr "Reproducir"
 
 #: src/pbview.cpp:372
+#. XXX FIXME --
```
Looks like a note to yourself that you forgot to remove?

Minoru commented on po/es.po:
```diff
@@ -777,34 +774,32 @@ msgid "Description: "
-#, fuzzy
 msgid "Saving bookmark on autopilot..."
-msgstr "Guardando marcados..."
+msgstr "Guardando marcador automáticamente... "
```
This still has a trailing space, please remove it.

Minoru commented on po/es.po:
```diff
@@ -947,12 +940,12 @@ msgstr "Artículo guardado en %s"
 #, c-format
 msgid "Search Result - '%s'"
-msgstr "Resultado de la búsqueda - «%s»"
+msgstr "Resultado de la búsqueda - '%s'"
```
Sorry for wading into your territory, but this looks inconsistent: you're using angle quotes for the search string on line 687, but here you use single quotes.

Please ignore this comment if there's an actual rationale behind this.

Minoru commented:
Hmm, my review got split into three. I updated one of my comments since then, please take a look at https://github.com/newsboat/newsboat/pull/874#discussion_r410436525 if you're reading this through email.

programandala-net commented:
@Minoru

>  msgid "audio"
>  msgstr ""
> 
> Did you forget to add a translation here? You translated "video" above, after all.

No translation was needed in this case, it's the same word. But I've
added it now, to make it clear it's not forgotten.

> +#. XXX FIXME --
> Looks like a note to yourself that you forgot to remove?

Yes, thanks.

-- 
Marcos Cruz
http://programandala.net

programandala-net commented:
@Minoru

> +msgstr "Guardando marcador automáticamente... "
> 
> This still has a trailing space, please remove it.

Thanks. It seems somehow it wasn't catched by the regexp I used.

> -msgstr "Resultado de la búsqueda - «%s»"
> +msgstr "Resultado de la búsqueda - '%s'"
> 
> Sorry for wading into your territory, but this looks inconsistent:
> you're using angle quotes for the search string on line 687, but here
> you use single quotes.
> 
> Please ignore this comment if there's an actual rationale behind this.

Thank you for the observation. You are right. The fact is the work is
not finished. I started to homogenize the usage of quotes in the
translation. In Spanish there's no need to mark URLs or filenames, they
can be unquoted, though in this context angles could be used for
clarity. As for code, in this context, any usual convention is fine,
e.g. `x`. As for actual text quotes, angle quotes are the norm for the
first nesting level.

But I soon realized the usage in the original wasn't consistent either:
'x', `x', `x` and simply x are used for text, code, filenames and URLs.
Apparently several conventions have been inherited from previous
versions of the program. And the meaning of several quotes was not clear
to me without consulting the sources. Therefore I decided to stop the
task on the quotes and open a new issue later. I added a small note in
the commit message.

-- 
Marcos Cruz
http://programandala.net

programandala-net commented:
>  #: newsboat.cpp:122
> -#. An actual filename, no translation needed
> +#. TRANSLATORS: This is the 'config' filename description used in a table.
> 
> This is going to be removed the next time someone runs `make
> msgmerge`. Please add these comments to the source code. They should
> start with three forward slashes for gettext to pick them up.

OK. I've searched for other examples in the sources. It's clear. But at
the moment I can not be sure whether my local <src/newsboat.cpp> is
updated. What do you recommend?

> For these comments to be added to .po files, you'll have to run `make
> extract && make msgmerge`, but please don't do that: it will create
> conflicts with current master branch. I'll do this after merging your
> PR, and then you can start your next PR with an updated version.

Thanks for the explanation.

>  #: src/configcontainer.cpp:305
>  msgid "%N %V - URLs"
> -msgstr "%N %V - URLs"
> +msgstr "%N %V - URL"
> 
> Withdrawing this because titles don't need articles, see
> https://github.com/newsboat/newsboat/pull/874#issuecomment-612441172.

Sorry, not sure what you mean. The form "URL" (instead of the old
translation "URLs") is right, as we already discussed in that comment.

-- 
Marcos Cruz
http://programandala.net

Minoru commented:
> Apparently several conventions have been inherited from previous versions of the program.

Most probably no-one ever gave it much thought, just like punctuation (#879). So you'll file an issue about that, right?

> at the moment I can not be sure whether my local <src/newsboat.cpp> is updated. What do you recommend?

You can check by fetching the latest commits (without merging them into your branch) and comparing the files:

    $ git fetch upstream
    $ git diff upstream/master master -- newsboat.cpp

In general, I'd just make a commit and push it here. If there is a conflict, GitHub will tell us, and then we can work on rebasing your changes onto current master.

> > Withdrawing this because titles don't need articles, see #874 (comment).
>
> Sorry, not sure what you mean. […]

That's just a comment for a future reader (potentially, me) who wonders why certain review comments were never addressed. You don't need to do anything about this one ;)

Minoru commented:
Will you file that issue about the use of quotes, @programandala-net?

programandala-net commented:
@Minoru

> Looks great! Thanks for pushing this over the finish line, @programandala-net!

Thank you for your help.

I reviewed the full text again and improved some style details. Probably
the usage will reveal issues that were difficult to see out of context.
Also the shortening of some keyboard legends is pending. I'll see what
can be done, after trying this version.

-- 
Marcos Cruz
http://programandala.net

Minoru commented:
I missed one important CI failure: [i18nspector pointed out](https://travis-ci.com/github/newsboat/newsboat/jobs/321856730#L262) that the `Last-Translator` field is "wrong". Removing the site address didn't help, so I added the email that you use in your commits. That works: https://github.com/newsboat/newsboat/commit/452a6dcb8c3aba0ff71acaaf08a5e526c15733fd If you want to list some other email there, just submit a PR.

I also re-generated the POT file to include things that were added while you were working on this PR. When you start working on a new set of changes, please branch off https://github.com/newsboat/newsboat/commit/452a6dcb8c3aba0ff71acaaf08a5e526c15733fd or some later commit.

> Will you file that issue about the use of quotes, @programandala-net?

Never mind this, I filed https://github.com/newsboat/newsboat/issues/901 myself.

programandala-net commented:
> the `Last-Translator` field is "wrong". Removing the site address
> didn't help, so I added the email that you use in your commits. That
> works:

> If you want to list some other email there, just submit a PR.

OK, thanks. I didn't supposed an email is mandatory in that field. In
fact I used my domain instead, as a means of contact, to prevent spam.
Maybe I'll change it as you say, following your advice about branches.

> Never mind this, I filed
> https://github.com/newsboat/newsboat/issues/901 myself.

Thank you. I didn't forget it, but not everyday I can dedicate some time
to this.

-- 
Marcos Cruz
http://programandala.net
